### PR TITLE
Title component for download section

### PR DESCRIPTION
**Pull request by alex6061**

### CODE REVIEW

File: js/components/widgets/Title.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @function Title
+ * @typedef {Object} Title
+ * @property {string} priority
+ * @param {data} title 
+ * @param {string} parrentClassName 
+ * @returns {string} HTML or empty
+ */
+
+export const Title = (data, parrentClassName) => {
+  if(!Title) return '';
```
Comment (nagorapro) [inline js/components/widgets/Title.js:11]: Title is your function name
You don't have the Title parameter!!!

File: js/components/widgets/Title.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @function Title
+ * @typedef {Object} Title
+ * @property {string} priority
+ * @param {data} title
```
Comment (nagorapro) [inline js/components/widgets/Title.js:5]: Title is your function name
You don't have the Title parameter!!!

File: js/components/widgets/Title.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   `${parrentClassName}__title` :
+    'title';
+
+  switch(priority){
+    default: return `<h1 class="${currentClassName}>${content}</h1>"`;
+    case 2: return `<h1 class="${currentClassName}>${content}</h1>"`;
+    case 3: return `<h1 class="${currentClassName}>${content}</h1>"`;
+    case 4: return `<h1 class="${currentClassName}>${content}</h1>"`;
+    case 5: return `<h1 class="${currentClassName}>${content}</h1>"`;
+    case 6: return `<h1 class="${currentClassName}>${content}</h1>"`;
```
Comment (nagorapro) [inline js/components/widgets/Title.js:25]: Always <h1>?

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Title.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+export const Title = (data, parrentClassName) => {
+  if(!Title) return '';
+
+  const { priority, content } = data;
+
+  const currentClassName = parrentClassName ?
+   `${parrentClassName}__title` :
+    'title';
```
Comment (nagorapro) [inline js/components/widgets/Title.js:17]: const currentClassName = parrentClassName
  ? `${parrentClassName}__title`
  : 'title';

File: js/components/widgets/Title.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  switch(priority){
```
Comment (nagorapro) [inline js/components/widgets/Title.js:19]: switch(priority) SPACE {

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Title.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @function Title
+ * @typedef {Object} Title
+ * @property {string} priority
```
Comment (nagorapro) [inline js/components/widgets/Title.js:4]: Put the definition of your custom Title type in a separate JSDoc comment
Your Title custom type should have two properties: priority and e.g. text

File: js/components/widgets/Title.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * @param {data} data
```
Comment (nagorapro) [inline js/components/widgets/Title.js:5]: * @param {Title} data

File: js/components/widgets/Title.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * @function Title
+ * @typedef {Object} Title
+ * @property {string} priority
+ * @param {data} data 
+ * @param {string} parrentClassName 
+ * @returns {string} HTML or empty
+ */
+
+export const Title = (data, parrentClassName) => {
+  if(!data) return '';
```
Comment (nagorapro) [inline js/components/widgets/Title.js:11]: if (!Object.keys(data).length === 2) return '';

File: js/components/widgets/Title.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+  const { priority, content } = data;
+
+  const currentClassName = parrentClassName 
+    ? `${parrentClassName}__title`
+    : 'title';
+
+  switch(priority) {
```
Comment (nagorapro) [inline js/components/widgets/Title.js:19]: switch SPACE (priority)

File: js/components/widgets/Title.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    default: return `<h1 class="${currentClassName}>${content}</h1>"`;
+    case 2: return `<h2 class="${currentClassName}>${content}</h2>"`;
+    case 3: return `<h3 class="${currentClassName}>${content}</h3>"`;
+    case 4: return `<h4 class="${currentClassName}>${content}</h4>"`;
+    case 5: return `<h5 class="${currentClassName}>${content}</h5>"`;
+    case 6: return `<h6 class="${currentClassName}>${content}</h6>"`;
+  }
```
Comment (nagorapro) [inline js/components/widgets/Title.js:26]: ;

File: js/components/widgets/Title.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+}
```
Comment (nagorapro) [inline js/components/widgets/Title.js:27]: ;

File: js/components/widgets/Title.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  if(!data) return '';
+
+  const { priority, content } = data;
+
+  const currentClassName = parrentClassName 
+    ? `${parrentClassName}__title`
+    : 'title';
+
+  switch(priority) {
+    default: return `<h1 class="${currentClassName}>${content}</h1>"`;
```
Comment (nagorapro) [inline js/components/widgets/Title.js:20]: Your line: default: return `<h1 class="${currentClassName}>${content}</h1>"`;
My line: default: return `<h1 class="${currentClassName}">${content}</h1>`;

File: js/components/widgets/Title.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    case 2: return `<h2 class="${currentClassName}>${content}</h2>"`;
```
Comment (nagorapro) [inline js/components/widgets/Title.js:21]: Your line: case 2: return `<h2 class="${currentClassName}>${content}</h2>"`;
My line: case 2: return `<h2 class="${currentClassName}">${content}</h2>`;

File: js/components/widgets/Title.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+    case 3: return `<h3 class="${currentClassName}>${content}</h3>"`;
+    case 4: return `<h4 class="${currentClassName}>${content}</h4>"`;
+    case 5: return `<h5 class="${currentClassName}>${content}</h5>"`;
+    case 6: return `<h6 class="${currentClassName}>${content}</h6>"`;
```
Comment (nagorapro) [inline js/components/widgets/Title.js:25]: Same here

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Title.js (hunk 6 above)
Comment (alex6061) [reply js/components/widgets/Title.js:4]: What do you mean by custom title?

Comment (nagorapro) [reply js/components/widgets/Title.js:4]: That's the type you came up with on your own

Comment (nagorapro) [reply js/components/widgets/Title.js:4]: * @typedef {Object} Title
* @property {number} priority
* @property {string} content

File: js/components/widgets/Title.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * @function Title
+ * @param {Title} data 
+ * @param {string} parrentClassName 
+ * @returns {string} HTML or empty
+ */
+
+export const Title = (data, parrentClassName) => {
+  if (!Object.keys(data).length === 2) return '';
```
Comment (nagorapro) [inline js/components/widgets/Title.js:15]: if (Object.keys(data).length !== 2) return '';

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Title.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
+/**
+ * @property {number} priority
+ * @property {string} content
+ */
+
+/**
+ * @function Title
+ * @typedef {Object} Title
```
Comment (nagorapro) [inline js/components/widgets/Title.js:8]: ???

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Title.js (hunk 16 above)
Comment (alex6061) [reply js/components/widgets/Title.js:8]: Why are there two comments?

(I did not fully understand the dependence, what comes from)

File: js/components/widgets/Title.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+ * @function Title
+ * @param {Title} data 
+ * @param {string} parrentClassName 
+ * @returns {string} HTML or empty
+ */
+
+export const Title = (data, parrentClassName) => {
+  if (Object.keys(data).length !== 2) return '';
+
+  const { priority, content } = data.title;
```
Comment (nagorapro) [inline js/components/widgets/Title.js:17]: Bad

const { priority, content } = data;

**Review by nagorapro — COMMENTED** [review]
Need fix

**Pull request merged**